--- src/mux_h1.c	2021-07-07 15:46:09.000000000 +0100
+++ src/mux_h1.updated.c	2021-08-04 12:29:07.000000000 +0100
@@ -1396,20 +1396,6 @@
 		goto end;
 	}
 
-	/* If websocket handshake, search for the websocket key */
-	if ((h1m->flags & (H1_MF_CONN_UPG|H1_MF_UPG_WEBSOCKET)) ==
-	    (H1_MF_CONN_UPG|H1_MF_UPG_WEBSOCKET)) {
-		int ws_ret = h1_search_websocket_key(h1s, h1m, htx);
-		if (!ws_ret) {
-			h1s->flags |= H1S_F_PARSING_ERROR;
-			TRACE_ERROR("missing/invalid websocket key, reject H1 message", H1_EV_RX_DATA|H1_EV_RX_HDRS|H1_EV_H1S_ERR, h1s->h1c->conn, h1s);
-			h1_capture_bad_message(h1s->h1c, h1s, h1m, buf);
-
-			ret = 0;
-			goto end;
-		}
-	}
-
 	if (h1m->err_pos >= 0)  {
 		/* Maybe we found an error during the parsing while we were
 		 * configured not to block on that, so we have to capture it
